ckanext/emailasusername/templates/user/new_user_form.html

```diff
@@ -2,8 +2,30 @@
 
 <form id="user-register-form" action="" method="post">
   {{ form.errors(error_summary) }}
-  {{ form.input("name", id="field-username", label=_("Username"), placeholder=_("username"), value=data.name, error=errors.name, classes=["control-medium"], is_required=True) }}
-  {{ form.input("fullname", id="field-fullname", label=_("Full Name"), placeholder=_("Joe Bloggs"), value=data.fullname, error=errors.fullname, classes=["control-medium"]) }}
+
+  <div id="username-input-field">
+    {{ form.input("fullname", id="field-fullname", label=_("Full Name"), placeholder=_("Joe Bloggs"), value=data.fullname, error=errors.fullname, classes=["control-medium"], attrs={'data-module': 'slug-preview-target', 'class': 'form-control'}, is_required=True) }}
+    {% set attrs = {'data-module': 'slug-preview-slug', 'data-module-placeholder': '<username>', 'class': 'form-control'} %}
+    {{ form.prepend('name', id='field-username', label=_('Username'), placeholder=_('username'), value=data.name, error=errors.name, attrs=attrs, is_required=true) }}
+  </div>
+  <style>
+    /* change URL: field to Username: */
+    #username-input-field .slug-preview strong {
+        display: none;
+    }
+    #username-input-field .slug-preview:before {
+        content: 'Username:';
+        font-weight: bold;
+    }
+    /* make username input conform to form style */
+    #username-input-field .input-group{
+      width: 100%;
+    }
+    #username-input-field .form-control{
+      border-radius: 3px;
+    }
```

Comment:
Before:
![image](https://user-images.githubusercontent.com/2634482/135253593-0ab1ed7d-2e80-4eac-a2f5-05bc5839bfa7.png)
By default, the input field is too small in this form

After:
![image](https://user-images.githubusercontent.com/2634482/135253678-da386462-3c92-480a-a848-1f851ff2ec71.png)
Some simple css helps fix this